catalog/dags/maintenance/update_science_museum_urls.py

```diff
@@ -0,0 +1,164 @@
+"""
+# Update Science Museum URLs
+
+One-time maintenance DAG to update Science Museum records to have valid URLs. See https://github.com/WordPress/openverse/issues/4261.
+
+For each Science Museum record, this DAG:
+
+* updates the url to the new format, excluding `/images/` in the path if it exists
+* validates whether the url . If not, the record ID is added to an `invalid_science_musem_ids` table.
+
+Once complete, we can use the `invalid_science_museum_ids` to identify records to delete. They are not automatically deleted by this DAG, in order to give us an opportunity to first see how many there are.
+"""
+
+import logging
+from datetime import timedelta
+from textwrap import dedent
+
+from airflow.decorators import dag, task
+from airflow.exceptions import AirflowSkipException
+from airflow.models.abstractoperator import AbstractOperator
+from airflow.models.param import Param
+
+from common import slack
+from common.constants import DAG_DEFAULT_ARGS, POSTGRES_CONN_ID
+from common.sql import RETURN_ROW_COUNT, PostgresHook
+from common.urls import rewrite_redirected_url
+
+
+logger = logging.getLogger(__name__)
+
+
+DAG_ID = "update_science_museum_urls"
+INVALID_IDS_TABLE = "science_museum_invalid_ids"
+
+UPDATE_URLS_QUERY = dedent(
+    """
+    UPDATE image
+    SET url=REPLACE(url, '/images', '')
+    WHERE provider='sciencemuseum' AND url ILIKE '%/images/%';
+    """
+)
+CREATE_TABLE_QUERY = dedent(
+    f"""
+    CREATE TABLE IF NOT EXISTS {INVALID_IDS_TABLE}
+    (identifier uuid);
+    """
+)
+
+
+@task
+def run_sql(
+    sql: str,
+    handler: callable = None,
+    postgres_conn_id: str = POSTGRES_CONN_ID,
+    task: AbstractOperator = None,
+):
+    postgres = PostgresHook(
+        postgres_conn_id=postgres_conn_id,
+        default_statement_timeout=PostgresHook.get_execution_timeout(task),
+    )
+
+    return postgres.run(sql, handler=handler)
+
+
+@task
+def get_batches(
+    batch_size: int,
+    postgres_conn_id: str = POSTGRES_CONN_ID,
+    task: AbstractOperator = None,
+):
+    postgres = PostgresHook(
+        postgres_conn_id=postgres_conn_id,
+        default_statement_timeout=PostgresHook.get_execution_timeout(task),
+    )
+    records = postgres.get_records(
+        "SELECT identifier, url FROM image where provider='sciencemuseum';"
+    )
+
+    return [records[i : i + batch_size] for i in range(0, len(records), batch_size)]
```

Comment:
True! We could also reduce the batch size (Science Museum doesn't have that many records, something like 120k). Do you have a preference?

Reply:
120k still seems like a lot to put in XComs 😅 I think I'd def prefer the temp table approach!

Reply:
I meant it wouldn't take too long to do very small batches because the total isn't so high, not that 120k is itself not much -- particularly since this can all be deleted as soon as it's done. But I've gone ahead and put them in a temp table.

Reply:
Ah, got it! And you're right, I hadn't even thought about potentially deleting the XComs after the run 🤦🏼‍♀️ I'm not used to the one-off DAG paradigm haha.